### PR TITLE
Provide a chapter on adding a request ID

### PR DESCRIPTION
**Pull request by weierophinney**
Based on a question I answered in Slack, this provides a recipe for adding a unique request ID that can then be tracked in your logs.

### CODE REVIEW

File: docs/book/v4/cookbook/provide-a-request-id.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+Request identifiers are usually generated at the web-server level.
+When you use a dedicated web servers such as Apache or nginx, a load balancer, or a reverse proxy, these can be configured to create and inject a request ID before it reaches your application.
+However, when using mezzio-swoole, the request handler runner we create is your web server.
+It has listeners that take care of logging, which means that the request generated must _already_ have the identifier if you want to be able to log it.
+
+This poses a problem: normally you will use middleware to propagate changes to the request.
+How can you do it at the Swoole web server level?
+
+The answer is deceptively simple: you can provide a [delegator factory](https://docs.mezzio.dev/mezzio/v3/features/container/delegator-factories/) for decorator on thethe service that converts the Swoole HTTP request instance into the equivalent [PSR-7](https://www.php-fig.org/psr/psr-7/) HTTP request instance that is then passed to your application.
```
Comment (gsteel) [inline docs/book/v4/cookbook/provide-a-request-id.md]: There's a double "the" on this line.

File: docs/book/v4/cookbook/provide-a-request-id.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+If you add it via middleware, however, it is not present in your access logs.
+
+## The solution
+
+Request identifiers are usually generated at the web-server level.
+When you use a dedicated web servers such as Apache or nginx, a load balancer, or a reverse proxy, these can be configured to create and inject a request ID before it reaches your application.
+However, when using mezzio-swoole, the request handler runner we create is your web server.
+It has listeners that take care of logging, which means that the request generated must _already_ have the identifier if you want to be able to log it.
+
+This poses a problem: normally you will use middleware to propagate changes to the request.
```
Comment (froschdesign) [inline docs/book/v4/cookbook/provide-a-request-id.md:13]: Until here the problem is described, therefore the the headline "The solution" is wrong before these lines.

File: docs/book/v4/cookbook/provide-a-request-id.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+How can you do it at the Swoole web server level?
+
+The answer is deceptively simple: you can provide a [delegator factory](https://docs.mezzio.dev/mezzio/v3/features/container/delegator-factories/) for decorator on the service that converts the Swoole HTTP request instance into the equivalent [PSR-7](https://www.php-fig.org/psr/psr-7/) HTTP request instance that is then passed to your application.
```
Comment (froschdesign) [inline docs/book/v4/cookbook/provide-a-request-id.md]: Maybe it is _simple_ for you but it does not mean that it applies to others as well. So do not use "simple" or "easy" in the documentation.

**Review by froschdesign — CHANGES_REQUESTED** [review]

File: docs/book/v4/cookbook/provide-a-request-id.md (hunk 2 above)
Comment (weierophinney) [reply docs/book/v4/cookbook/provide-a-request-id.md:13]: Updated.

File: docs/book/v4/cookbook/provide-a-request-id.md (hunk 3 above)
Comment (weierophinney) [reply docs/book/v4/cookbook/provide-a-request-id.md]: Updated.

**Pull request merged**